packages/flutter/lib/src/services/raw_keyboard_macos.dart

```diff
@@ -0,0 +1,341 @@
+    final int independentModifier = modifiers & deviceIndependentMask;
+    switch (key) {
+      case ModifierKey.controlModifier:
+        return _isLeftRightModifierPressed(side, independentModifier & modifierControl, modifierLeftControl, modifierRightControl);
+      case ModifierKey.shiftModifier:
+        return _isLeftRightModifierPressed(side, independentModifier & modifierShift, modifierLeftShift, modifierRightShift);
+      case ModifierKey.altModifier:
+        return _isLeftRightModifierPressed(side, independentModifier & modifierOption, modifierLeftOption, modifierRightOption);
+      case ModifierKey.metaModifier:
+        return _isLeftRightModifierPressed(side, independentModifier & modifierCommand, modifierLeftCommand, modifierRightCommand);
+      case ModifierKey.capsLockModifier:
+        return independentModifier & modifierCapsLock != 0;
+      case ModifierKey.numLockModifier:
+        return independentModifier & modifierNumericPad != 0;
+      case ModifierKey.functionModifier:
+      case ModifierKey.symbolModifier:
+      case ModifierKey.scrollLockModifier:
+        // These are not used in macOS keyboards.
+        return false;
+    }
+    return false;
+  }
+
+  @override
+  KeyboardSide getModifierSide(ModifierKey key) {
+    KeyboardSide findSide(int leftMask, int rightMask) {
+      final int combinedMask = leftMask | rightMask;
+      final int combined = modifiers & combinedMask;
+      if (combined == leftMask) {
+        return KeyboardSide.left;
+      } else if (combined == rightMask) {
+        return KeyboardSide.right;
+      } else if (combined == combinedMask) {
+        return KeyboardSide.all;
+      }
+      return null;
+    }
+
+    switch (key) {
+      case ModifierKey.controlModifier:
+        return findSide(modifierLeftControl, modifierRightControl);
+      case ModifierKey.shiftModifier:
+        return findSide(modifierLeftShift, modifierRightShift);
+      case ModifierKey.altModifier:
+        return findSide(modifierLeftOption, modifierRightOption);
+      case ModifierKey.metaModifier:
+        return findSide(modifierLeftCommand, modifierRightCommand);
+      case ModifierKey.capsLockModifier:
+      case ModifierKey.numLockModifier:
+      case ModifierKey.scrollLockModifier:
+      case ModifierKey.functionModifier:
+      case ModifierKey.symbolModifier:
+        return KeyboardSide.all;
+    }
+
+    assert(false, 'Not handling $key type properly.');
+    return null;
+  }
+
+  // Modifier key masks. Apple's NSEvent documentation
+  // https://developer.apple.com/documentation/appkit/nseventmodifierflags?language=objc
+  // https://opensource.apple.com/source/IOHIDFamily/IOHIDFamily-86/IOHIDSystem/IOKit/hidsystem/IOLLEvent.h.auto.html
+
+  /// This mask is used to check the [modifiers] field to test whether the CAPS
+  /// LOCK modifier key is on.
+  ///
+  /// Use this value if you need to decode the [modifiers] field yourself, but
```

Comment:
This paragraph could be a dartdoc template:
```dart
  /// {@template flutter.services.logicalKeyboardKey.modifiers}
  /// Use this value if you need to decode the [modifiers] field yourself, but
  /// it's much easier to use [isModifierPressed] if you just want to know if
  /// a modifier is pressed.
  /// {@endtemplate}
```

And then replace each of the identical paragraphs with:
```dart
  /// {@macro flutter.services.logicalKeyboardKey.modifiers}
```

Reply:
Great, thanks!